src/main/java/net/coderbot/iris/pipeline/ClearPassCreator.java

```diff
@@ -75,17 +75,23 @@ public static ImmutableList<ClearPass> createClearPasses(RenderTargets renderTar
 		return ImmutableList.copyOf(clearPasses);
 	}
 
-	public static ImmutableList<ClearPass> createShadowClearPasses(ShadowRenderTargets renderTargets,
+	public static ImmutableList<ClearPass> createShadowClearPasses(ShadowRenderTargets renderTargets, boolean fullClear,
 																   PackShadowDirectives renderTargetDirectives) {
 		List<ClearPass> clearPasses = new ArrayList<>();
 
 		for (int i = 0; i < renderTargets.getNumColorTextures(); i++) {
+			// TODO: Refractor this to not depend on different numbers and color buffer numbers.
 			if (i < renderTargetDirectives.getColorSamplingSettings().size()) {
 				if (renderTargetDirectives.getColorSamplingSettings().get(i).getClear()) {
 					PackShadowDirectives.SamplingSettings samplingSettings = renderTargetDirectives.getColorSamplingSettings().get(i);
 					clearPasses.add(new ClearPass(samplingSettings.getClearColor(), renderTargets::getResolution, renderTargets::getResolution, renderTargets.getFramebufferForColorTexture(i), GL21C.GL_COLOR_BUFFER_BIT | GL21C.GL_DEPTH_BUFFER_BIT));
 				} else {
-					clearPasses.add(new ClearPass(new Vector4f(1.0F), renderTargets::getResolution, renderTargets::getResolution, renderTargets.getFramebufferForColorTexture(i), GL21C.GL_DEPTH_BUFFER_BIT));
+					if (fullClear) {
+						PackShadowDirectives.SamplingSettings samplingSettings = renderTargetDirectives.getColorSamplingSettings().get(i);
+						clearPasses.add(new ClearPass(samplingSettings.getClearColor(), renderTargets::getResolution, renderTargets::getResolution, renderTargets.getFramebufferForColorTexture(i), GL21C.GL_COLOR_BUFFER_BIT | GL21C.GL_DEPTH_BUFFER_BIT));
+					} else {
```

Comment:
This seems to be a copy of the other branch... Why not just do:

```java
if (fullClear || renderTargetDirectives.getColorSamplingSettings().get(i).getClear()) {
```

Reply:
Fixed.